lib/features/auth/view/widget/auth_input_field.dart

```diff
@@ -0,0 +1,108 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:telware_cross_platform/core/theme/palette.dart';
+
+class AuthInputField extends ConsumerStatefulWidget {
```

Comment:
as long as you didn't use 'ref' inside the widget, then use a normal StatefulWidget.
when ref is used, then we will convert it to ConsumerStatefulWidget.